### PR TITLE
present-cli: init at 0.2.2

### PR DESCRIPTION
**Pull request by cameronfyfe**
###### Description of changes

A script interpolation engine for markdown documents.

This tool makes it easy to incorporate generated cli content into a README markdown document.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/present-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "terror";
+    repo = pname;
+    rev = version;
+    sha256 = "+kCHe84ikdCLd7j5YwP2j3xz+XTzzo/kLy+b9YUFDnI=";
+  };
+
+  cargoSha256 = "VKY/FQUrFWtLxKoK6LP6qPMqNN4absZvnAbH9mha1fI=";
+
+  doCheck = false;
```
Comment (NickCao) [inline pkgs/tools/misc/present-cli/default.nix:19]: Please add a comment about why tests are disabled.

Comment (cameronfyfe) [reply pkgs/tools/misc/present-cli/default.nix]: When I looked into why tests were failing I realized it was just an unpatched shebang so fixed that and enabled tests.

File: pkgs/tools/misc/present-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "+kCHe84ikdCLd7j5YwP2j3xz+XTzzo/kLy+b9YUFDnI=";
+  };
+
+  cargoSha256 = "VKY/FQUrFWtLxKoK6LP6qPMqNN4absZvnAbH9mha1fI=";
+
+  # required for tests
+  preBuild = ''
+    patchShebangs bin/get_version
```
Comment (NickCao) [inline pkgs/tools/misc/present-cli/default.nix:21]: This is normally done in the `postPatch` phase.

Comment (cameronfyfe) [reply pkgs/tools/misc/present-cli/default.nix:21]: Thanks, updated.

**Review by NickCao — APPROVED** [review]

**Pull request merged**